tools/sslsniff_example.txt

```diff
@@ -103,10 +103,53 @@ lot of characters that are not printable or even Unicode replacement
 characters.
 
 
+Function latency can be show by -l or --latency option, and SSL handshake calls
+can be trace by --handshake option. This is useful when analyzing SSL/TLS
+performance. Tracing output of "openssl s_client -connect example.com:443":
+
+# ./sslsniff.py -l --handshake
+FUNC         TIME(s)            COMM             PID     LEN    LAT(ms)
+WRITE/SEND   0.000000000        openssl          3260509 0      N/A
+WRITE/SEND   0.593772992        openssl          3260509 0      593.773
```

Comment:
This is super confusing. I would like to merge this two. You can use similar mechanism to SSL_read in bpf programs by only emitting data on uretprobe. The same hashtable `bufs` can be used for SSL_write as well as I think in the same thread, there is no interleaving for SSL_read and SSL_write.

Reply:
Agreed on this. I'm focusing on OpenSSL, so didn't want to change previous uprobes for the other two SSL/TLS library, GnuTLS and NSS. I'll try to merge the SSL_write events and re-use bufs sniffing from SSL_read.